src/app/surveys/surveys.component.ts

```diff
@@ -145,36 +151,61 @@ export class SurveysComponent implements OnInit, AfterViewInit, OnDestroy {
   deleteSelected() {
     const selected = this.selection.selected.map(surveyId => findByIdInArray(this.surveys.data, surveyId));
     if (selected.length === 1) {
-      const survey = selected[0];
-      this.openDeleteDialog(this.deleteSurvey(survey), 'single', survey.name);
+       const survey = selected[0];
+       this.openDeleteDialog(this.deleteSurvey(survey), 'single',
+       survey.name + ( survey.course ? ' will be deleted from ' + survey.course.courseTitle + ' aswell' : '' ));
     } else {
       this.openDeleteDialog(this.deleteSurveys(selected), 'many', '');
     }
   }
 
+  createCourseArray(surveys) {
+    const courseArr = [];
+    const surveyCourse = surveys.filter((survey) => survey.course);
+    surveyCourse.forEach((element: any) => {
+      const indexAtArr = courseArr.findIndex((mCourse) => mCourse._id === element.course._id);
+      const course = indexAtArr > -1 ? courseArr[indexAtArr] : element.course;
+      if (element.stepNum > -1) {
+        delete course.steps[element.stepNum]['survey'];
+      }
+      indexAtArr === -1 ? courseArr.push(course) : courseArr[indexAtArr] = course;
+    });
+    return courseArr;
+  }
+
   deleteSurveys(surveys) {
     const deleteArray = createDeleteArray(surveys);
+    const courseArray = this.createCourseArray(surveys);
     return {
       request: this.couchService.bulkDocs(this.dbName, deleteArray),
       onNext: () => {
         this.surveys.data = this.surveys.data.filter((survey: any) => findByIdInArray(deleteArray, survey._id) === undefined);
         this.selection.clear();
         this.deleteDialog.close();
         this.planetMessageService.showMessage('You have deleted ' + deleteArray.length + ' surveys');
+        if (courseArray.length > 0) {
+          this.couchService.bulkDocs('courses', courseArray).subscribe(() => console.log('course removed'));
+        }
       },
       onError: () => this.planetMessageService.showAlert('There was a problem deleting survey.')
     };
   }
 
   deleteSurvey(survey) {
     const { _id: surveyId, _rev: surveyRev } = survey;
+    if (survey.course) {
+      delete survey.course.steps[this.findSurveyInSteps(survey.course.steps, survey)]['survey'];
```

Comment:
Two things on this line:
- We shouldn't delete from the survey variable directly.  This is a reference to the data object in the table, so if the request fails the data in the table would be updated without the database changing.  In this situation it's not bad since this property isn't shown in the table, but it's more future proof if we avoid this completely.
- You've only completed half of my earlier comment.  This is the other half, where rather than calling `findSurveyInSteps` again you can just reference `survey.stepNum`